pzsd_bot/__main__.py

```diff
@@ -187,6 +191,98 @@ async def leaderboard(ctx):
     await ctx.respond(embed=embed)
 
 
+@bot.slash_command(description="Add new name that can be bestowed points.")
+@option("name", description="The exact name to use when bestowing points.")
+@option("snowflake", description="Their discord ID if applicable.", required=False)
+@default_permissions(administrator=True)
+async def register(ctx, name, snowflake):
+    logger.info(
+        "%s invoked /register with name=%s and snowflake=%s",
+        ctx.author.name,
+        name,
+        snowflake,
+    )
+
+    async with engine.connect() as conn:
+        result = await conn.execute(select(pzsd_user).where(pzsd_user.c.name == name))
+
+    user_to_add = result.one_or_none()
+    if user_to_add is not None:
+        if user_to_add.is_active:
+            logger.info("User '%s' already exists", name)
+            await ctx.respond(f"'{name}' already exists!")
+            return
+        else:
+            logger.info("User '%s' exists but is inactive", name)
+            async with engine.begin() as conn:
+                await conn.execute(
+                    update(pzsd_user)
+                    .where(pzsd_user.c.name == name)
+                    .values(is_active=True, discord_snowflake=snowflake)
+                )
+    else:
+        async with engine.begin() as conn:
+            await conn.execute(
+                insert(pzsd_user).values(
+                    name=name,
+                    discord_snowflake=snowflake,
+                )
+            )
+
+    logger.info("Added user '%s' to user table", name)
+
+    await ctx.respond(f"Added user with name {name}")
+
+
+@bot.slash_command(
+    description="Remove name from being able to be bestowed points.",
+)
+@option("name", description="The exact name in the user table")
+@default_permissions(administrator=True)
+async def unregister(ctx, name):
+    logger.info(
+        "%s invoked /unregister with name=%s",
+        ctx.author.name,
+        name,
+    )
+
+    async with engine.connect() as conn:
+        result = await conn.execute(select(pzsd_user).where(pzsd_user.c.name == name))
+
+    user_to_del = result.one_or_none()
+    if user_to_del is None or not user_to_del.is_active:
+        logger.info("User '%s' doesn't exist in user table", name)
```

Comment:
Log message not true in inactive case

Reply:
I had the same logic for this log as I explained in your other comment. This one is maybe more confusing though, I could be open to changing it.